coordinator/persistence-api/src/main/java/org/apache/cassandra/stargate/transport/ProtocolVersion.java

```diff
@@ -84,19 +90,60 @@ public enum ProtocolVersion implements Comparable<ProtocolVersion> {
           && Boolean.parseBoolean(System.getProperty("stargate.enable_dse_protocol_v2", "false"));
 
   /** All supported versions */
-  public static final EnumSet<ProtocolVersion> SUPPORTED =
-      supportDseProtocol ? EnumSet.of(V3, V4, V5, DSE_V1, DSE_V2) : EnumSet.of(V3, V4, V5);
+  public static final EnumSet<ProtocolVersion> SUPPORTED;
 
-  public static final List<String> SUPPORTED_VERSION_NAMES =
-      SUPPORTED.stream().map(ProtocolVersion::toString).collect(Collectors.toList());
+  /** The preferred versions */
+  public static final ProtocolVersion CURRENT;
+
+  public static final Optional<ProtocolVersion> BETA;
+
+  private static final String MAX_PROTOCOL_VERSION_PROP = "stargate.max_protocol_version";
+
+  static {
+    int maxProtocolVersion;
+    try {
+      maxProtocolVersion = Integer.getInteger(MAX_PROTOCOL_VERSION_PROP, Integer.MAX_VALUE);
+    } catch (NumberFormatException e) {
+      throw new ConfigurationException(
+          "Invalid integer value for "
+              + MAX_PROTOCOL_VERSION_PROP
+              + ": "
+              + System.getProperty(MAX_PROTOCOL_VERSION_PROP));
+    }
+    if (maxProtocolVersion < 1)
```

Comment:
The condition needs to be (maxProtocolVersion < 3), so there is at least 1 supported version. Supported OS versions are {V3, V4, V5, V6}.